src/lang/KclSingleton.ts

```diff
@@ -401,6 +401,8 @@ export class KclManager {
     this.errors = errors
     // Do not add the errors since the program was interrupted and the error is not a real KCL error
     this.addDiagnostics(isInterrupted ? [] : kclErrorsToDiagnostics(errors))
+    // Add warnings and non-fatal errors
+    this.addDiagnostics(complilationErrorsToDiagnostics(execState.errors))
```

Comment:
I'm pretty sure we'll want this like above. When the user switches files, it can interrupt long-running programs by causing the promises/awaited engine responses to fail early. This can create lots of errors that aren't real and shouldn't be shown.

```suggestion
    this.addDiagnostics(isInterrupted ? [] : complilationErrorsToDiagnostics(execState.errors))
```